src/io/hdf/hdf_wrapper_functions.h

```diff
@@ -133,6 +133,25 @@ inline bool h5d_read(hid_t grp, const std::string& aname, T* first, hid_t xfer_p
   return ret != -1;
 }
 
+template<typename T>
+inline bool h5d_check_type(hid_t grp, const std::string& aname)
+{
+  if (grp < 0)
+    return true;
+  hid_t h1 = H5Dopen(grp, aname.c_str(), H5P_DEFAULT);
+  if (h1 < 0)
+    return false;
+  T temp;
+  hid_t h5d_type_id     = get_h5_datatype(temp);
+  hid_t datatype        = H5Dget_type(h1);
```

Comment:
Should we check that `H5Dget_type` doesn't return `H5I_INVALID_HID`?
https://docs.hdfgroup.org/hdf5/develop/group___h5_d.html#title25

Reply:
I believe the above return if the dataset tag doesn't exist prevent us from encountering `H5I_INVALID_HID`. However I like assuming that the tag should exist + throwing an error if something goes wrong along the way since this is more explicit. Fixed